x/distribution/abci_app.go

```diff
@@ -0,0 +1,25 @@
+package distribution
+
+import (
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/distribution/keeper"
+)
+
+// set the proposer for determining distribution during endblock
+func BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock, k keeper.Keeper) {
+	consAddr := sdk.ConsAddress(req.Header.ProposerAddress)
+	k.SetProposerConsAddr(ctx, consAddr)
+
+	// XXX TODO actually calculate this
+	k.SetSumPrecommitPower(ctx, sdk.NewDec(1))
```

Comment:
Maybe we should pass this through from staking, where we iterate over precommits anyways

Reply:
did you mean slashing?

Reply:
I added in the calculation - I'm not sure it's really going to be more efficient to do this in the slashing cycle / make the code more like spaghetti, so I suggest we leave this seperate in this PR and then maybe refactor it together after this PR